contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -236,4 +264,23 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
         // Calls NativeMinter precompile through INativeMinter interface.
         NATIVE_MINTER.mintNativeCoin(recipient, adjustedAmount);
     }
+
+    /**
+     * @dev Scales `value` based on `tokenMultiplier` and the direction of the transfer.
+     * Should be used for all tokens being transferred to/from other subnets.
+     */
+    function _scaleTokens(uint256 value, bool isReceive) private view returns (uint256) {
+        if (multiplyOnReceive == isReceive) {
```

Comment:
Can you add a comment above this conditional to help clarify the xor check? Something like "Scale multiplicitavely if we're scaling on the same side of the transfer as the contract instance multiplies". Then below, a comment like "Otherwise, scale by dividing the value by the multiplier".

Reply:
It's a matter of preference, but I think this function would be more readable (even at the cost of being more verbose) by restructuring the control flow to explicitly check the 4 possible cases in an `if-else` tree.